addons/actions.md

```diff
@@ -7,8 +7,89 @@ title: Actions
+
+When the result of the commandString is to cause some action (e.g. turn on a light, change a thermostat to a new temperature, etc.) then use sendCommand. When interacting with widgets on the user interfaces commands are sent to the Items.
+
+When the result of the commandString is to change the state of an Item without causing some action (i.e. to make the state of an Item match the state of a device without commanding the device to change).
+
+As a general rule, is better to call `MyItem.sendCommand(command)` and `MyItem.postUpdate(command)` where possible because the Item methods are able to handle a wider variety of commands appropriately. The Actions are best reserved for use in cases where the Item's name is determined at runtime.
+
+* `Map<Item, State> storeStates(Item item1, Item item2, ... Item itemn)`: Returns a `Map<Item, State>` with the current state of each Item. All members of Groups are put into the Map but not the Group's state itself.
+
+* `restoreStates(Map<Item, State> statesMap)`: Restores the items' states from the map. If the saved state can be interpreted as a command (ON/OFF/etc.), a command is sent to the Item. Otherwise an update is sent to the Item.
+
+### Audio Actions
+
+* `setMasterVolume(float volume)`: Sets the volume of the host machine (volume in range 0-1)
+* `increaseMasterVolume(float percent)`: Increases the volume by the given percent
+* `decreaseMasterVolume(float percent)`: Decreases the volume by the given percent
+* `float getMasterVolume()`: Returns the current volume as a float between 0 and 1
+* `playSound(String filename)`: Plays the given sound file. The file must be an mp3 or wav and located in `${openhab.home}/sounds`
+* `playStream(String url)`: Plays the audio stream at the given url
+* `say(String text)`: Says the given text through Text-to-Speech
+* `say(String text, String voice)`: Says the given text through the given voice (depends on the TTS engine and voices installed)
+* ` say(String text, String voice, String device)`: Says the given text through the given voice to the given output device (Mac OSX only).
+
+To get a list of available voices use `say -v ?` and to get a list of devices uses `say -a ?`.
+
+### Logging Actions
+
+* `logDebug(String loggername, String logText)`: Logs logText on level DEBUG to openhab.log
+* `logInfo(String loggername, String logText)`: Logs logText on level INFO to openhab.log
+* `logWarn(String loggername, String logText)`: Logs logText on level WARN to openhab.log
+* `logError(String loggername, String logText)`: Logs logText on level ERROR to openhab.log
+
+`logText` can be a compete String, constructed through concatination, or through arguments.
+
+**Complete String Example:** `logInfo("Garage", "This is a complete String")`
+**Concatination Example:** `logDebug("Lighting", "This is a string concatination:" + Light.name)`
+**Arguments Example:** `logWarn("Alarm", "The {} window has been open for {} hours!", Window.name, timeOpen)`
```

Comment:
These will be joined without newlines. Either itemize them or force a break at the end of the lines with `<br />`